### ISSUE TITLE
FTBFS 2.1.0

### ISSUE THREAD

**Issue by jengelh** [libsemigroups/libsemigroups#320]
Targeting i386-linux-gnu,

```
abuild@a4:~/rpmbuild/BUILD/libsemigroups-2.1.0> make V=1
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I./config  -DNDEBUG      -I/home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/include -I/home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/extern -std=gnu++14 -O3 -Wall -Wextra   -I/usr/include/eigen3     -fomit-frame-pointer -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -c -o src/libsemigroups_la-cong.lo `test -f 'src/cong.cpp' || echo './'`src/cong.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I./config -DNDEBUG -I/home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/include -I/home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/extern -std=gnu++14 -O3 -Wall -Wextra -I/usr/include/eigen3 -fomit-frame-pointer -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -c src/cong.cpp  -fPIC -DPIC -o src/.libs/libsemigroups_la-cong.o
In file included from /home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/include/libsemigroups/kambites.hpp:60,
                 from /home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/include/libsemigroups/cong.hpp:29,
                 from src/cong.cpp:22:
/home/abuild/rpmbuild/BUILD/libsemigroups-2.1.0/include/libsemigroups/string-view.hpp:406:35: error: static assertion failed: sizeof(Short) must equal sizeof(Long)
  406 |       static_assert(sizeof(Short) == sizeof(Long),
      |                     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
make: *** [Makefile:3018: src/libsemigroups_la-cong.lo] Error 1
```

```
(gdb) ptyp l
class libsemigroups::detail::StringViewContainer::Long {
  private:
    std::vector<libsemigroups::detail::StringView, std::allocator<libsemigroups::detail::StringView> > _views;
...
}
(gdb) ptyp s
type = class libsemigroups::detail::StringViewContainer::Short {
  private:
    libsemigroups::detail::StringViewContainer::const_iterator_string _begin1;
    libsemigroups::detail::StringViewContainer::const_iterator_string _begin2;
    uint32_t _length1;
    uint32_t _length2;
    ...
}
```

That static_assert makes no sense. There simply is more stuff in Short than in Long and how big they will be is anyone's guess (in particular because std::vector<>'s size is dependent upon the implementation).

**Comment by james-d-mitchell** [comment]
Agreed, I should really introduce some 32 bit CI jobs, will fix asap

**Comment by james-d-mitchell** [comment]
Hopefully resolved in v2.1.1, thanks for the report!
